File: CS214-Programming Language Concepts/projects/03/Ruby/grades.rb
#grades.rb computes letter grade of given int class average
# Input:int average
# Output: the letter grade associated with average
#
# By: Joshua Maguire
# Date:2/26/14
###############################################################


#computeGrages outputs letter grade coresponding to recieved int
# Input:int average
# Output: the letter grade associated with average
def computeGrades(myAverage)
	case (myAverage/10)
		when 9..10 then puts 'A' #if myAverage is in range 9 to 10 after being divided by 10, then put "A".
		when 8..9 then puts 'B'#^^^...
		when 7..8 then puts 'C'
		when 6..7 then puts 'D'
		else puts 'F'
	end		
end


if __FILE__ == $0 #If file run directly
	print "Enter int average:"
	average=gets.chomp.to_f #get the int from keyboard
	computeGrades(average) #run method, which outputs result.
end

File: CS214-Programming Language Concepts/projects/03/c++/grades.cpp
/* grades.cpp tests the letterGrade method given from the project. By Joshua Maguire on 2/26/14 for Calvin College.*/

#include <iostream>

using namespace std;

char letterGrade(int average);//declaring method


int main()
{
  cout << "testing letterGrade method...\n";
  cout << "average 100 = " << letterGrade(100) << "\n";
  cout << "average 90 = " << letterGrade(90) << "\n";
  cout << "average 80 = " << letterGrade(80) << "\n";
  cout << "average 77 = " << letterGrade(77) << "\n";
  cout << "average 61 = " << letterGrade(61) << "\n";
  cout << "average 55 = " << letterGrade(55) << "\n";
  cout << "average 111 = " << letterGrade(111) << "\n";
  cout << "testing complete.\n";
}


   char letterGrade(int average)
   {
      switch (average / 10)
      {
         case 10: case 9:
            return 'A';
         case 8:
            return 'B';
         case 7:
            return 'C';
         case 6:
            return 'D';
         default:
            return 'F';
      }
   }
